orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/states/SaltStateService.java

```diff
@@ -159,13 +159,21 @@ private Optional<String> extractJidIfPossible(SaltExecutionWentWrongException e)
 
     private Multimap<String, Map<String, String>> applyStateJidInfo(SaltConnector sc, String jid) {
         JidInfoResponse jidInfo = sc.run("jobs.lookup_jid", RUNNER, JidInfoResponse.class, "jid", jid);
+        if (jidInfo.isEmpty()) {
+            LOGGER.error("jobs.lookup_jid returns an empty response: {}", jidInfo);
+            throw new SaltExecutionWentWrongException("jobs.lookup_jid returns an empty response. Please check the salt log.");
+        }
         LOGGER.debug("Salt apply state jid info: {}", jidInfo);
         Map<String, List<RunnerInfo>> states = JidInfoResponseTransformer.getSimpleStates(jidInfo);
         return collectMissingTargets(states);
     }
 
     private Multimap<String, Map<String, String>> highStateJidInfo(SaltConnector sc, String jid) {
         JidInfoResponse jidInfo = sc.run("jobs.lookup_jid", RUNNER, JidInfoResponse.class, "jid", jid, "missing", "True");
+        if (jidInfo.isEmpty()) {
```

Comment:
Hi @akanto yes! Same as other exceptions this will have a retry as well in the `doCall()` method under `OrchestratorBootstrapRunner.iava`, the number of retry will depend on the number of maximum retry on error. Below is an example of the log when it is retrying on this error. And yes, it should be temporary unless salt or the node is completely down. In our case, for example, it only returns an empty response there's no space on the disk, after our code removes those temporary files and the disk is freed up, the response will be back again. The only concern I have is if we have any functions that specifically will have an empty response (which will conflict with the purpose of lookup jid though since lookup jid should return something). 
`Orchestrator component SaltJobIdTracker failed to start, retrying [12/180], error count [6/20]. Elapsed time: 2600 ms, Total elapsed time: 136519 ms, Reason: com.sequenceiq.cloudbreak.orchestrator.exception.CloudbreakOrchestratorFailedException: jobs.lookup_jid returns an empty response. Please check the salt log., additional info: `